narwhal/primary/src/state_handler.rs

```diff
@@ -1,35 +1,163 @@
 // Copyright (c) 2021, Facebook, Inc. and its affiliates
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
-use config::AuthorityIdentifier;
+use config::{AuthorityIdentifier, Committee};
+use crypto::RandomnessPrivateKey;
+use fastcrypto::groups;
+use fastcrypto_tbls::dkg;
 use mysten_metrics::metered_channel::{Receiver, Sender};
 use mysten_metrics::spawn_logged_monitored_task;
 use tap::TapFallible;
 use tokio::task::JoinHandle;
 use tracing::{debug, error, info, warn};
-use types::{Certificate, CertificateAPI, ConditionalBroadcastReceiver, HeaderAPI, Round};
+use types::{
+    Certificate, CertificateAPI, ConditionalBroadcastReceiver, HeaderAPI, Round, SystemMessage,
+};
 
-/// Receives the highest round reached by consensus and update it for all tasks.
+type G = groups::bls12381::G2Element; // TODO-DNS is it correct for these to be the same?
```

Comment:
yes, for now, but let's leave them as two types since we might change the ECIES key type

Reply:
perhaps name those BEACON_PK_G and BEACON_ENC_G?